cmd/injective-price-oracle/oracle.go

```diff
@@ -305,3 +295,55 @@ func oracleCmd(cmd *cli.Cmd) {
 		closer.Hold()
 	}
 }
+
+func NewCosmosClient(ctx context.Context, senderAddress cosmtypes.AccAddress, cosmosKeyring keyring.Keyring, network common.Network, cosmosConfig *CosmosConfig) (chainclient.ChainClient, error) {
+	if cosmosConfig != nil {
+		if cosmosConfig.tendermintRPC != "" {
+			network.TmEndpoint = cosmosConfig.tendermintRPC
+		}
+		if cosmosConfig.cosmosGRPC != "" {
+			network.ChainGrpcEndpoint = cosmosConfig.cosmosGRPC
+		}
+		if cosmosConfig.cosmosStreamGRPC != "" {
+			network.ChainStreamGrpcEndpoint = cosmosConfig.cosmosStreamGRPC
+		}
+	}
+
+	clientCtx, err := chainclient.NewClientContext(network.ChainId, senderAddress.String(), cosmosKeyring)
+	if err != nil {
+		log.WithError(err).Fatalln("failed to initialize cosmos client context")
+	}
+
+	tmRPC, err := rpchttp.New(network.TmEndpoint, "/websocket")
+	if err != nil {
+		log.WithError(err).Fatalln("failed to connect to tendermint RPC")
+	}
+
+	clientCtx = clientCtx.WithNodeURI(network.TmEndpoint).WithClient(tmRPC)
+	txFactory := chainclient.NewTxFactory(clientCtx)
+	txFactory = txFactory.WithGasAdjustment(cosmosConfig.cosmosGasAdjust)
+	txFactory = txFactory.WithGasPrices(cosmosConfig.cosmosGasPrices)
+
+	cosmosClient, err := chainclient.NewChainClient(clientCtx, network, common.OptionTxFactory(&txFactory))
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
+			"endpoint": network.ChainGrpcEndpoint,
+		}).Fatalln("failed to connect to daemon, is injectived running?")
+	}
```

Comment:
The flow is pretty mixed up. On some cases like this we stop the app with `fatalln`. On others we just log.
Maybe we need to move all error handling on `NewCosmosClient` to just log errors and, if we are using more than one cosmos clients, if we dont have at least 1 available we stop the app 
OR
we just return an error and if there's any error, we stop.

I would prefer if we stop when even one fails. Cosmos clients are not supposed to fail so we should also fail at startup